File: arrays_and_hashing/two_sum.py
class Solution:
    def twoSum(self, nums, target):
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in nums[i+1:]: 
                return [i, nums.index(diff, i+1)]

solution = Solution()
print(solution.twoSum([3,2,4], 6))

File: two_pointers/two_sum.js
function twoSum(numbers, target) {
    let hash = new Map();
    for (let i = 0; i < numbers.length; i++) {
        let diff = target - numbers[i];
        let curr = numbers[i]

        if (hash.has(curr)) {
            return [hash.get(curr), i + 1]
        } else {
            hash.set(diff, i + 1)
        }
    }
};

console.log(twoSum([2, 7, 11, 15], 9));
